functions/configfile.pp

```diff
@@ -0,0 +1,5 @@
+function mcollective::configfile(Hash $confighash) >> String {
+  #$result = ["# Configuration is managaged by puppet class $name"]
+  $result = $confighash.map |$k, $v| { "${k} = ${v}" }
+  (["# Configuration is managaged by puppet class ${name}"] << $result << []).join("\n")
```

Comment:
> we'll have to sort by the keys to make sure hash ordering changing does not cause repeated restarts of daemons

I think there is no need: Keys should in the same order and should only change on parameter change.